text/0000-project-error-handling.md

```diff
@@ -0,0 +1,137 @@
+- Feature Name: error_handling_project_group
+- Start Date: 2020-07-23
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC establishes a new project group, under the libs team, to drive efforts to improve error handling in Rust.
+
+# Motivation
+[motivation]: #motivation
+
+The error handling project group aims to reduce confusion on how to structure error handling for users in the Rust community. This will be accomplished by creating learning resources and pushing effort to upstream widely used crates into the standard library. As a secondary goal, this project group will also try to resolve some known issues with the `Error` trait and reporting errors in panics/termination.
+
+# Charter
+[charter]: #charter
+
+## Goals
+
+### Agree on and define common error handling terminology
+
+- Recoverable error: An error that can reasonably be expected to be encountered e.g. a missing file.
+- Unrecoverable error: An error that cannot reasonably be expected to happen and which indicates a bug e.g. indexing out of bounds.
```

Comment:
Hm, why define "recoverable" and "unrecoverable" errors by how *likely* they're thought to be rather than by whether it's feasible for software to recover from them?

In particular, the subsequent recommendation to use "`Result` and `Error` types for recoverable errors" and "`panic` for unrecoverable errors" would seem, at least to me, to agree better with the latter, more literal interpretation of the terms "recoverable" and "unrecoverable".